// brackets-typescript/dist/node/tslint-utils.js
"use strict";
var _ = require('lodash');
function mapLintResultFailures(failures) {
    return (failures || []).map(function (failure) {
        return {
            type: 'problem_type_warning',
            message: failure.failure + ' [' + failure.ruleName + ']',
            pos: {
                line: failure.startPosition.lineAndCharacter.line,
                ch: failure.startPosition.lineAndCharacter.character
            }
        };
    });
}
exports.mapLintResultFailures = mapLintResultFailures;
function executeTsLint(fullPath, code, TSLint, tsLintConfig, program) {
    try {
        var options = {
            configuration: tsLintConfig
        };
        var tsLinter = new TSLint(fullPath, code, options, program);
        var result = tsLinter.lint();
        return _.sortBy(mapLintResultFailures(result.failures), function (item) { return item.pos.line; });
    }
    catch (err) {
        return [{
                type: 'problem_type_error',
                message: "TSLintError: " + err.toString(),
                pos: { line: 0, ch: 0 }
            }];
    }
}
exports.executeTsLint = executeTsLint;
